base/params/list.go

```diff
@@ -1,7 +1,25 @@
 // Package params defines generic input parameter structures
 package params
 
-import "fmt"
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+var (
+	// ErrListParamsNotEmpty indicates that there are list values in the params
+	ErrListParamsNotEmpty = fmt.Errorf("list params not empty")
+	// DefaultListLimit is the default value for a list
+	DefaultListLimit = 25
+)
+
+// ListParams can pull list param information from an http.Request
+type ListParams interface {
+	// ListParamsFromRequest pulls list params from the URL
+	ListParamsFromRequest(r *http.Request) error
```

Comment:
The only issue I have is with this interface. I don't think it's strictly necessary, and may just be more confusing than useful. But, I don't have any strong suggestions currently on how to replace it, so it's not a big deal.